lib/core/src/lib/datatable/components/datatable/datatable.component.ts

```diff
@@ -780,10 +784,18 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
     }
 
     markRowAsContextMenuSource(selectedRow: DataRow): void {
+        this.selectedRowId = selectedRow.id ? selectedRow.id : '';
         this.data.getRows().forEach((row) => row.isContextMenuSource = false);
         selectedRow.isContextMenuSource = true;
     }
 
+    private setRowAsContextSource(): void {
```

Comment:
you are running this on `ngOnChanges`, that means hundreds of times per second potentially. you can call this from the `markRowAsContextMenuSource` as this looks like a one-time operation. You already set the `selectedRowId` there, so why not updating the row?

Reply:
or use the get/set for the `selectedRowId` to perform this 1 time only when needed. Please avoid polluting ngOnChanges with stuff that does not need to run a lot of times

Reply:
Its running only two times. And it needs to be executed in ngOnChanges and not in markRowAsContextMenuSource because the data gets refreshed after the operation of restore in manage versions is performed, markRowAsContextMenuSource is called before the said operation.

Reply:
I understand that it needs to be run multiple times, but in your case you run it on *each* property change, so the performance will have a hit. Check the ngOnChanges docs, each time one of the properties will change it state, you will be looping through the data and performing the same task again.